theia/templates/statefulset.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "theia.fullname" . }}
+  labels:
+    {{- include "theia.labels" . | nindent 4 }}
+spec:
+  serviceName: "{{ template "theia.fullname" . }}"
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "theia.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        dev.okteto.com/inject-token: "true"
+      labels:
+        {{- include "theia.selectorLabels" . | nindent 8 }}
+    spec:
+      volumes:
+        - name: cloudbin
+          hostPath: 
+            path: /var/okteto/cloudbin
+            type: Directory
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "theia.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: ssh-key
+          image: "{{ template "theia.image" . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: 
+          - sh 
+          - -c
+          - test -f /var/okteto/ssh/id_rsa && cat /var/okteto/ssh/id_rsa.pub && exit 0; ssh-keygen -q -t rsa -N '' -f '/var/okteto/ssh/id_rsa' && cat /var/okteto/ssh/id_rsa.pub
+          volumeMounts:
+          - name: data
+            mountPath: /var/okteto/ssh
+            subPath: ssh
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ template "theia.image" . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: 
+          - sh 
+          - -c
+          - (mkdir -p $HOME/.ssh && chmod 700 $HOME/.ssh && cp /var/okteto/ssh/* $HOME/.ssh/ && chmod 600 $HOME/.ssh/id_rsa*); {{ if .Values.repository }}(test -d /home/project/.git || git clone {{ .Values.repository }} /home/project);{{ end }}PATH=$PATH:/var/okteto/bin node $HOME/src-gen/backend/main.js /home/project --hostname=0.0.0.0
```

Comment:
I would clone using a github token instead. I think it simplifies the flow for private repositories (you could even set the TOKEN as an okteto secret, it would available for all your deployments.

Reply:
What do you mean by clone with a token?

Reply:
git clone https://<token>@github.com/owner/repo.git

Reply:
Ah I see. But what is the benefit in that approach vs the keys? (Keys are more standard no?)

Reply:
https is now the recommended way:
https://docs.github.com/en/github/using-git/which-remote-url-should-i-use#cloning-with-https-urls-recommended